src/App.jsx

```diff
@@ -1,44 +1,82 @@
-import { useState } from "react"
-import {initialWorkouts, generateWorkout} from "./Workouts.js"
-import "./App.css"
+import { useState } from "react";
+import { initialWorkouts, generateWorkout } from "./Workouts.js";
+import "./App.css";
 
 function App() {
-  const [workouts, setWorkouts] = useState(initialWorkouts)
+  const [workouts, setWorkouts] = useState(initialWorkouts);
 
   const addNewWorkout = () => {
-    const newWorkout = generateWorkout()
-    console.log("addNewWorkout:", newWorkout)
-  }
+    const newWorkout = generateWorkout();
+    console.log("addNewWorkout:", newWorkout);
+    const addedWorkout = [...workouts, newWorkout];
+    setWorkouts(addedWorkout);
+  };
 
   const deleteWorkout = (workout) => {
-    console.log("deleteWorkout:", workout)
-  }
+    console.log("deleteWorkout:", workout);
+    const updatedWorkouts = workouts.filter(function (workouts) {
+      return workouts !== workout;
+    });
+    setWorkouts(updatedWorkouts);
+  };
 
   const completeWorkout = (workout) => {
-    console.log("completeWorkout:", workout)
-  }
+    console.log("completeWorkout:", workout);
+    const updatedWorkouts = workouts.map(function (workouts) {
+      if (workouts === workout) {
+        return { ...workouts, done: true };
+      }
+      return workouts;
+    });
+    setWorkouts(updatedWorkouts);
+  };
+
+  const randomiseWorkout = (workout) => {
+    console.log("oldWorkout:", workout);
+    const updatedWorkout = generateWorkout();
+    const updatedWorkouts = workouts.map(function (workouts) {
```

Comment:
your variable naming is wrong, the inner function uses a variable name `workouts` when it should be a singular variable; then this workouts variable name clashes with the existing workouts variable state....

better to rename it to `w` or `existingWorkout`